lib/mime-type.js

```diff
@@ -68,6 +68,37 @@ module.exports = class MIMEType {
     return serialize(this);
   }
 
+  isJS() {
```

Comment:
This should check that there are no parameters present, as they are prohibited by the HTML Standard.

Reply:
thats not true, see https://github.com/whatwg/html/pull/3205 , the only restriction is on the `type` parameter of a `<script>` tag, which does not map to MIME (hence use of `type=module`.

Reply:
The linked PR only applies to Content-Type of the response, not e.g. the HTML `type` attribute. As it is, none of the [JavaScript MIME types](https://html.spec.whatwg.org/multipage/scripting.html#javascript-mime-type) contain parameters.

Reply:
> For external JavaScript resources, MIME type parameters in `Content-Type` headers are generally ignored. (In some cases the `charset` parameter has an effect.)

Reply:
For the purposes of this module, which is used by jsdom, we should stick to the actually-defined JS MIME type list, not the vague Content-Type behavior that is outside the scope of HTML.

Reply:
@domenic doesn't https://html.spec.whatwg.org/multipage/webappapis.html#fetch-a-single-module-script also check for the MIME, not just string matches? My usage and reason for this PR is a service worker trying to detect if a Module is coming over the wire not parsing script tag attributes.

Reply:
Yeah, I guess the HTML spec uses both checks. I suppose we should have an `{ excludeParameters = false } = {}` parameter list for this.

Reply:
I'm open to renaming, but was confused on the exact nature of it being named `excludeParameters` excluding it from the check or ensuring they were no parameters. I've called it `allowParameters` in a new change.